File: applications/luci-statistics/luasrc/model/cbi/luci_statistics/interface.lua
--[[

Luci configuration model for statistics - collectd interface plugin configuration
(c) 2008 Freifunk Leipzig / Jo-Philipp Wich <xm@leipzig.freifunk.net>

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

        http://www.apache.org/licenses/LICENSE-2.0

$Id$

]]--

require("luci.sys")


m = Map("luci_statistics",
	translate("Interface Plugin Configuration"),
	translate(
		"The interface plugin collects traffic statistics on " ..
		"selected interfaces."
	))

-- collectd_interface config section
s = m:section( NamedSection, "collectd_interface", "luci_statistics" )

-- collectd_interface.enable
enable = s:option( Flag, "enable", translate("Enable this plugin") )
enable.default = 0

-- collectd_interface.interfaces (Interface)
interfaces = s:option( MultiValue, "Interfaces", translate("Monitor interfaces") )
interfaces.widget = "select"
interfaces.size   = 5
interfaces:depends( "enable", 1 )
for k, v in pairs(luci.sys.net.devices()) do
	interfaces:value(v)
end

-- collectd_interface.ignoreselected (IgnoreSelected)
ignoreselected = s:option( Flag, "IgnoreSelected", translate("Monitor all except specified") )
ignoreselected.default = 0
ignoreselected:depends( "enable", 1 )

return m

File: applications/luci-app-statistics/htdocs/luci-static/resources/view/statistics/plugins/interface.js
'use strict';
'require baseclass';
'require form';
'require tools.widgets as widgets';

return baseclass.extend({
	title: _('Interface Plugin Configuration'),
	description: _('The interface plugin collects traffic statistics on selected interfaces.'),

	addFormOptions: function(s) {
		var o;

		o = s.option(form.Flag, 'enable', _('Enable this plugin'));

		o = s.option(widgets.DeviceSelect, 'Interfaces', _('Monitor interfaces'));
		o.multiple = true;
		o.noaliases = true;
		o.default = 'br-lan';
		o.depends('enable', '1');

		o = s.option(form.Flag, 'IgnoreSelected', _('Monitor all except specified'));
		o.depends('enable', '1');
	},

	configSummary: function(section) {
		var ifaces = L.toArray(section.Interfaces),
		    invert = section.IgnoreSelected == '1';

		if (ifaces.length == 0)
			return _('Monitoring all interfaces');
		else if (invert)
			return N_(ifaces.length, 'Monitoring all but one interface', 'Monitoring all but %d interfaces').format(ifaces.length);
		else
			return N_(ifaces.length, 'Monitoring one interface', 'Monitoring %d interfaces').format(ifaces.length);
	}
});
